src/twisted/names/newsfragments/9918.misc

```diff
@@ -0,0 +1 @@
+New interface IEncodableRecord combines IEncodable and IRecord, which is useful when using type annotations.
```

Comment:
Newsfragments which are misc need to be empty.  See:

https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles

Can you separate out the changes where you introduce a IEncodableRecord interface into a new PR, and leave this .misc file as empty?

Everything else in this PR is very simple in terms of adding return types to `__repr__()` and are easy to merge in.

Reply:
Done: `IEncodableRecord` and the other `dns` changes were split off to #1362